### ISSUE TITLE
[Suggestion] Locking Compacting Drawers to accept only compatible items

### ISSUE THREAD

**Issue by bswan** [jaquadro/StorageDrawers#399]
I love this mod very-very-much! Especialy with other mods that let's me have remote acces to my storage arrays. Bad thing about this bit: I need to manualy select what items I want to put to my compacting drawers cause in other case they will be filled by any uncompatible stuff.

So idea is to have some way to prevent Compacting drawers from accepting wrong items.

It can be:
1) New kind of key (most preferrable)
2) New upgrade item (e.g. 'filter upgrade' but not actual item filter, just trigger to not accept junk)
3) Default behaviour
4) Configurable behaviour

Thanks for doing avesome job, man

**Comment by codewarrior0** [comment]
To clarify, are you asking for a way to prevent Compacting Drawers from storing an item that cannot be compacted? The Drawer Key already lets you lock the drawer to a single item (or to no items, preventing automation from inserting anything until you put in the first item by hand).

**Comment by bswan** [comment]
Yeah I want it to not accept items that can not be compacted, but accept any other one.

**Comment by bookerthegeek** [comment]
I fail to find a valid use case for this that can not be solved be either locking the drawer to the item, or building more drawers.

**Comment by bswan** [comment]
That's what I doing now: putting every new item to compacting drawer to check is that compatible, and if it is - locking it there, but I'd prefer more elegant way :)

**Comment by codewarrior0** [comment]
What about a use-case like this one: You are playing a new modpack with unfamiliar mods. You have hundreds of itemstacks in chests and you're not sure which of the new items can be compacted. You'd like to set up your drawer array, connect the chests to the controller with item pipes, and watch the compactable items occupy compacting drawers while the rest of the items fall into normal drawers.

GregTech in particular forbids metals from being compacted in hand or in drawers.
